src/gc/impl/manual/gc.d

```diff
@@ -35,46 +35,41 @@ static import core.memory;
 
 extern (C) void onOutOfMemoryError(void* pretend_sideffect = null) @trusted pure nothrow @nogc; /* dmd @@@BUG11461@@@ */
 
-class ManualGC : GC
+// register GC in C constructor (_STI_)
+extern(C) void _d_register_manual_gc()
 {
-    __gshared Array!Root roots;
-    __gshared Array!Range ranges;
-
-    static void initialize(ref GC gc)
-    {
-        import core.stdc.string;
+    import gc.registry;
+    registerGCFactory("manual", &initialize);
+}
 
-        if (config.gc != "manual")
-            return;
+private GC initialize()
+{
+    import core.stdc.string: memcpy;
 
-        auto p = cstdlib.malloc(__traits(classInstanceSize, ManualGC));
-        if (!p)
-            onOutOfMemoryError();
+    auto p = cstdlib.malloc(__traits(classInstanceSize, ManualGC));
+    if (!p)
+        onOutOfMemoryError();
 
-        auto init = typeid(ManualGC).initializer();
-        assert(init.length == __traits(classInstanceSize, ManualGC));
-        auto instance = cast(ManualGC) memcpy(p, init.ptr, init.length);
-        instance.__ctor();
+    auto init = typeid(ManualGC).initializer();
+    assert(init.length == __traits(classInstanceSize, ManualGC));
+    auto instance = cast(ManualGC) memcpy(p, init.ptr, init.length);
+    instance.__ctor();
 
-        gc = instance;
-    }
-
-    static void finalize(ref GC gc)
-    {
-        if (config.gc != "manual")
-            return;
+    return instance;
+}
 
-        auto instance = cast(ManualGC) gc;
-        instance.Dtor();
-        cstdlib.free(cast(void*) instance);
-    }
+class ManualGC : GC
+{
+    __gshared Array!Root roots;
+    __gshared Array!Range ranges;
```

Comment:
I know it's been this way before, but it doesn't really make sense to make these members `__gshared`.